backend/socket.io/listeners.js

```diff
@@ -1,8 +1,17 @@
 const clients = require('./clients');
+const channelValid = require('../api/chatLink/utils/validateChannel');
 
 const connectionListener = (socket, io) => {
-  socket.on('chat-join', (data) => {
+  socket.on('chat-join', async (data) => {
     const { userID, channelID, publicKey } = data;
+    try {
+      await channelValid(channelID);
```

Comment:
The [function](https://github.com/muke1908/chat-e2ee/blob/e08a23673a24cd41967c9d2d4dadb4ed1b39f0bf/backend/api/chatLink/utils/validateChannel.js#L4) `channelValid` returns an object like
```
return {
    valid: <true|false>,
    state: <state>
};
```
So it should be something like:
```
const { valid } = await channelValid(channelID);
if(!valid) {
    return
}
```

For reference, you can check this implementation 
https://github.com/muke1908/chat-e2ee/blob/e08a23673a24cd41967c9d2d4dadb4ed1b39f0bf/backend/api/messaging/index.js#L22

Reply:
yes, of course, thank you!